CONTRIBUTING.md

```diff
@@ -0,0 +1,380 @@
+bugs in multiple components. In such cases, use `bugfix:`. Other commonly used
+prefix words are:
+
+- `feature:` for features, or improvements, in multiple components
+- `ci:` for fixes or improvements in the CI process
+- `doc:` for fixes and improvements in the documentation
+
+These prefix words are for conventional purposes. Use common sense and make
+the commit message clear so that others can understand what the change is.
+
+The rest of the first line should start with a verb. Examples:
+
+```text
+foo: fix typos
+```
+
+```text
+foo: resolve race condition in bar()
+```
+
+The first line should make sense when reading _"When you merge this, it will
+`$THE_FIRST_LINE`"_.
+
+The second line of the commit message must be an empty line.
+
+In the rest of the commit message, write details of the change if necessary.
+Explain what it does _and_ *why*.  The lines in the commit message should be
+limited to 72 characters or less where possible.
+
+Include a reference to an Issue when the commit fixes an Issue.
+
+```text
+fixes #$ISSUE_NUMBER
+```
+
+When an Issue number or a Pull Request number is prefixed with certain
+keywords, the referenced Issue or Pull Request will be closed. See [Linking a
+pull request to an issue using a keyword](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)
+for the supported keywords.
+
+### Creating a Pull Request
+
+When your code is ready to be merged, and all the tests have passed in the CI,
+create a Pull Request. Before creating a Pull Request, make sure:
+
+1. You compiled the code and the build succeeded
+1. You pushed the changes to your remote fork and the CI passed
+1. Functions, macros, data types are documented in the code
+1. An example application is provided under [`examples`](examples)
+1. You compiled the example code and the example application ran on a
+   physical device as expected and documented
+1. All files are licensed under one of [Acceptable Licenses](#acceptable-licenses)
+   by including the license at the top of file
+1. One of your commits in the feature branch, or the PR itself, mentions Issue
+   number so that the Issue will be automatically closed when the PR is merged
+
+From this point, you should avoid to `git rebase` your feature branch.
+
+Developers who has write access to the repository will leave comments, ask
+rewrites, and merge the PR.
+
+## Licenses
+
+We provide code that can be freely used, copied, modified, and distributed by
+anyone and for any purpose.
+
+### Acceptable licenses
+
+We accept permissive licenses such as:
+
+- ISC License
+- MIT License
+- BSD License
+
+### Acceptable license for new code
+
+New code is the one you own (you wrote it from scratch).  The preferred
+license to be applied to new code is a simplified ISC License.  The license
+must be included at the top in every files as long as practically possible.
```

Comment:
> must be included at the top in every files

I'm not sure about this. I personally prefer to refer to LICENSE file in header, like
```C
/**
 * blah-blah-blah
 * 
 * MIT Licensed as described in the file LICENSE
 */

Reply:
some libraries may have files licensed under different terms. some files may have different copyright holders, or more copyright holders than others. it's also a common practice to audit license violations.

Reply:
But in this case, we will have to change the titles for all existing components.

Reply:
the case applies to new code. maybe the following wording is better?

> The license must be included at the top in every new files as long as practically possible.

or, just state preferred licenses and document how the license should be shown in the style guide?

and here is the definition of `new code` (from BSD [`style(9)`](https://man.openbsd.org/style)):

>     In general code can be considered “new code” when it makes up about 50%
>     or more of the file(s) involved.  This is enough to break precedents in
>     the existing code and use the current style guidelines.

Reply:
I think we should leave the document as it is and update the file headers that don't match. This will be more correct.

Reply:
the code style and handling of existing code aside (we will discuss further when the style guide is ready), are you okay with the choice of the acceptable licenses and the unacceptable licenses?

Reply:
> are you okay with the choice of the acceptable licenses and the unacceptable licenses?

Sure!